thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/client/pinot/summary/Cube.java

```diff
@@ -0,0 +1,91 @@
+package com.linkedin.thirdeye.client.pinot.summary;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+import org.apache.commons.lang3.builder.ToStringBuilder;
+import org.apache.commons.lang3.builder.ToStringStyle;
+
+import com.google.gson.Gson;
+
+
+// TODO: The cube should be queried from the backend database during runtime.
+public class Cube { // the cube (Ca|Cb)
+  private double ga; // sum of all ta
+  private double gb; // sum of all tb
+  private double rg; // global ratio, i.e., gb / ga
+  private List<Record> records = new ArrayList<>();
+
+  public int size() {
+    return records.size();
+  }
+
+  public Record get(int idx) {
+    return records.get(idx);
+  }
+
+  public void addRecord(Record record) {
+    this.records.add(record);
+  }
+
+  public void sort(Comparator<Record> comparator) {
+    Collections.sort(records, comparator);
+  }
+
+  public double getRg() {
+    return rg;
+  }
+
+  public void setRg(double rg) {
+    this.rg = rg;
+  }
+
+  public double getGa() {
+    return ga;
+  }
+
+  public void setGa(double ga) {
+    this.ga = ga;
+  }
+
+  public double getGb() {
+    return gb;
+  }
+
+  public void setGb(double gb) {
+    this.gb = gb;
+  }
+
+  public void toJson(String fileName) throws IOException {
+    FileWriter fw = new FileWriter(fileName);
```

Comment:
FileWriter is a Closeable resource. Please ensure to close it. If not closed, it may create leak.
Use "try with resources"